inbm/dispatcher-agent/dispatcher/sota/tiber_util.py

```diff
@@ -76,8 +79,14 @@ def tiber_download(dispatcher_broker: DispatcherBroker, uri: CanonicalUri,
         dispatcher_broker.telemetry(info_msg)
 
     try:
-        with requests.get(url=uri.value, headers=headers) as response:
-            repo.add(filename=file_name, contents=response.content)
+        with requests.get(url=uri.value, headers=headers, stream=True) as response:
+            response.raise_for_status()
+            with open(os.open(os.path.join(repo.get_repo_path(), file_name), os.O_CREAT | os.O_WRONLY), 'wb') \
+                    as destination_file:
+                for chunk in response.iter_content(chunk_size=16384):
+                    if cancel_event.is_set():
+                        raise SotaError("Download cancelled.")
```

Comment:
remove `destination_file` before raising error (be sure to surround with try/catch for `OSError` if necessary)

Reply:
Yes, when it raises the SotaError, there is a try/catch in sota.py to remove all the files inside the directory.
https://github.com/intel/intel-inb-manageability/blob/296682b2273dda7d1ca33dac99c1703e4465919c/inbm/dispatcher-agent/dispatcher/sota/sota.py#L363-L369

OSError is handled in try/catch statement. 
https://github.com/intel/intel-inb-manageability/blob/296682b2273dda7d1ca33dac99c1703e4465919c/inbm/dispatcher-agent/dispatcher/sota/tiber_util.py#L86-L91